src/matchmaking-threads.ts

```diff
@@ -0,0 +1,144 @@
+		matchmakingLockTimeout = 60 * 60;
+	}
+
+	const now = Date.now();
+	const matchmakingThreads = await MatchmakingThread.findAll();
+
+	for (const thread of matchmakingThreads) {
+		let threadChannel;
+		try {
+			threadChannel = await client.channels.fetch(thread.thread_id);
+		} catch (err) {
+			console.log(`Error fetching matchmaking thread channel: ${err}`);
+			threadChannel = null;
+		}
+
+		if (!threadChannel || threadChannel.type !== ChannelType.PublicThread) {
+			// * Thread is either deleted or not a public thread, which should not happen
+			console.log(`Removing deleted or invalid matchmaking thread from database: ${thread.thread_id}`);
+			await MatchmakingThread.destroy({ where: { thread_id: thread.thread_id } });
+			continue;
+		}
+
+		const lastMessageSent = new Date(thread.last_message_sent).getTime();
+		const timeSinceLastMessage = now - lastMessageSent;
+
+		if (timeSinceLastMessage > matchmakingLockTimeout * 1000) {
+			const guild = await threadChannel.guild.fetch();
+
+			// * Only send any inactivity messages if the thread has not already been manually closed/locked by moderators
+			if (!threadChannel.archived && !threadChannel.locked) {
+				if (threadChannel.ownerId) {
+					const creator = await User.findOne({ where: { user_id: threadChannel.ownerId } });
+					if (!creator?.matchmaking_notification_sent) {
+						const notificationEmbed = new EmbedBuilder();
+						notificationEmbed.setColor(0x6060ff);
+						notificationEmbed.setTitle('Your matchmaking thread has been automatically closed');
+						notificationEmbed.setDescription(
+							`Hello <@${threadChannel.ownerId}>! This is just to let you know that your Pretendo matchmaking thread <#${threadChannel.id}> has been automatically closed due to inactivity.\n\n` +
+								`**This is not a moderator action or punishment.** All threads in <#${threadChannel.parentId}> are automatically closed after a period of inactivity to ensure that each thread is dedicated to a single game session.\n\n` +
+								'**If you want to play again, just create a new thread and ping the people you want to invite. Have fun!**\n\n' +
+								'*Note:* This is a **one-time notification** that is sent to all server members to after their first matchmaking thread is closed. You will not be notified when your matchmaking threads are closed in the future.'
+						);
+						notificationEmbed.setFooter({
+							text: 'Thank you for using Pretendo!',
+							iconURL: guild.iconURL()!
+						});
+						const creatorUser = await client.users.fetch(threadChannel.ownerId);
+						try {
+							//TODO - Switch this to the new DM/notification channel system
+							await creatorUser.send({
+								embeds: [notificationEmbed]
+							});
+							await User.upsert({ user_id: threadChannel.ownerId, matchmaking_notification_sent: true });
+						} catch (err) {
+							console.log('Failed to DM user');
+						}
+					}
+				}
```

Comment:
Overall this whole chunk of code feels super claustrophobic. There's essentially no line spacings separating parts of the code. Typically we like to separate code with line breaks based on what they are/do. Take this snippet from the GitHub bot as an example:

```ts
if (issues.length !== 0) {
	const problems = increaseAndGetGitHubUserProblemsCount(payload.issue.user.id);
	let comment = 'Thank you for opening this issue! I have detected the following possible issues with your issue:\n';

	comment += '\n';

	for (const issue of issues) {
		comment += `- ${issue}\n`;
	}

	comment += '\n';
	comment += 'While we always appreciate users opening issues to try and help improve our services, it\'s important to follow the [Contribution Guidelines](https://github.com/PretendoNetwork/.github/blob/master/.github/CONTRIBUTING.md) to ensure smooth interactions!\n';
	comment += '\n';
	comment += 'Please refresh yourself on these guidelines for future issues!\n';

	// * Only show this after a certain amount of bad contributions, as to not scare new people
	if (problems <= 3) {
		comment += '\n';
		comment += `This is the ${ordinal(problems)} time I have detected errors in your contributions. Please be aware that continued instances of contributions with errors may result in future contributions being blocked.\n`;
	}

	comment += '\n';
	comment += '***Note: This process was automated. If you believe this was done in error, please ignore and consider filing a bug report on the [`webhook-listeners`](https://github.com/PretendoNetwork/webhook-listeners/issues/new/choose) repository.***';

	await octokit.issues.createComment({
		owner: payload.repository.owner.login,
		repo: payload.repository.name,
		issue_number: payload.issue.number,
		body: comment
	});
}
```

The variable declarations are grouped together due to them being similar in nature (with constants being defined first), then the modifications are separated and grouped as they are different from declarations but similar to themselves, and then checks like `if` and function calls like `await octokit.issues.createComment` are also separated

This goes for the whole codebase, but this chunk of code was the biggest example of it

Reply:
I think it's better now but please let me know if there are further improvements wanted.

Reply:
Looks a lot better now :+1: there's a couple little nitpicks I still have with it but nothing I think is worth holding the PR hostage over